### PR TITLE
Fix type errors in lwt examples

### PR DESCRIPTION
**Pull request by jplew**

### CODE REVIEW

File: examples/lwt/lwt_get_pipelined.ml [hunk 1]
```diff
@@ -33,18 +33,20 @@ let request host =
   in
   let* response = Client.get client "/" in
   let open Lwt_syntax.Async in
-  let* () =
+  let (stream, _) = Body.to_string_stream response.body in
+  let* () = 
     Lwt_stream.iter_s
```
Comment (anmonteiro) [inline examples/lwt/lwt_get_pipelined.ml:38]: If you're up to it, the recommended API to traverse the stream (and know whether the body was successfully read) is `Body.iter_string_s`: https://github.com/anmonteiro/piaf/blob/master/lib/piaf.mli#L236-L239

File: examples/lwt/lwt_get_pipelined.ml [hunk 2]
```diff
@@ -32,21 +32,11 @@ let request host =
       (Uri.of_string host)
   in
   let* response = Client.get client "/" in
-  let open Lwt_syntax.Async in
   let* () =
-    Lwt_stream.iter_s
-      (fun chunk -> Lwt_io.printf "%s" chunk)
-      (Body.to_string_stream response.body)
+    Body.iter_string_s (fun chunk -> Lwt_io.printf "%s" chunk) response.body
```
Comment (jplew) [inline examples/lwt/lwt_get_pipelined.ml:36]: @anmonteiro `Body.iter_string_s` returns the value of `or_error`, but it looks like we are swallowing the error here. Our main run function checks only for an error resulting from `Client.get client "/blog"`, and appears to ignore errors from `Client.get client "/"`. Is that correct?

File: examples/lwt/lwt_get_pipelined.ml (hunk 1 above)
Comment (jplew) [reply examples/lwt/lwt_get_pipelined.ml]: I used `Body.iter_string_s` in the latest commit, please have a look.

File: examples/lwt/lwt_get_pipelined.ml (hunk 2 above)
Comment (anmonteiro) [reply examples/lwt/lwt_get_pipelined.ml:36]: No, you'll only get there if there are no errors. It's subtle, but the `let*` is `Lwt_result.bind`.

**Comment by anmonteiro** [comment]
Thanks!

File: examples/lwt/lwt_get_pipelined.ml (hunk 2 above)
Comment (jplew) [reply examples/lwt/lwt_get_pipelined.ml:36]: Thanks for explaining. I'll have to investigate further where the process is bailing out.

Comment (anmonteiro) [reply examples/lwt/lwt_get_pipelined.ml:36]: It's not bailing out. It's equivalent to this:

```ocaml
let open Lwt_result.Infix in

Body.iter_string_s (fun chunk -> ...) >>= fun () ->
  request "/blog"
...
```

So you're only even issuing the second request if the first succeeded.

Comment (jplew) [reply examples/lwt/lwt_get_pipelined.ml:36]: I see! makes sense

**Pull request merged**